bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java

```diff
@@ -364,16 +364,16 @@ protected Set<BookieId> addDefaultRackBookiesIfMinNumRacksIsEnforced(
             Set<BookieId> bookiesInDefaultRack = null;
             Set<Node> defaultRackLeaves = topology.getLeaves(getDefaultRack());
             for (Node node : defaultRackLeaves) {
-                if (node instanceof BookieNode) {
+                if (node instanceof BookieNode && ((BookieNode) node).getAddr() != null) {
```

Comment:
When does node.getAddr() return null?
To prevent this, should I throw an error or use getDefaultRack?
Also, add a test case to verify this exception scenario.